### DISCUSSION TITLE
Connection to multiple LDAP servers

### DISCUSSION THREAD

**Discussion by mkk1001** [rabbitmq/rabbitmq-server#4164] [Questions] [answered]
Hi, how to connect RabbitMQ to two or more ldap servers (different domains)?

My configuration:
```
auth_backend.1 = internal
auth_backend.2 = ldap
auth_ldap.servers.1 = domain1.example.local
auth_ldap.timeout   = 10000
auth_ldap.port = 636
auth_ldap.use_ssl = true
```
It works for me
If I add another server ```auth_ldap.servers.2 = domain2.example.local```, i cant login

I tried use ${ad_domain} in config ``` auth_ldap.servers.1 = ${ad_domain}.example.local``` , this does not work

rabbitmq ver.3.8.26

**Comment by lukebakken** [comment, marked as answer]
Your configuration file is mis-formatted. Please see the docs - https://www.rabbitmq.com/ldap.html#connectivity

Note the `.` in use as a separator, **not** a `_` character:

```
auth_backend.1 = internal
auth_backend.2 = ldap
auth_ldap.servers.1 = domain1.example.local
auth_ldap.servers.2 = domain2.example.local
auth_ldap.timeout   = 10000
auth_ldap.port = 636
auth_ldap.use_ssl = true
```

Using a substitution like `${ad_domain}` is **not** supported in the `rabbitmq.conf` file.

**Reply by mkk1001** [reply]
I'm sorry my mistake.I have corrected the config in question

**Reply by mkk1001** [reply]
With this configuration, rabbitmq only sends requests to one server.
If i login `user1@domain1.example.local` request to `domain2.example.local`
why rmq not use all ldap.servers ?

**Reply by lukebakken** [reply]
When you specify multiple servers, it is used to round-robin connections and to provide a backup in case a connection fails. RabbitMQ and the LDAP plugin do *not* distinguish between servers based on the login credentials passed to the LDAP plugin.

**Reply by mkk1001** [reply]
thanks for help
